flows/actions/base.go

```diff
@@ -120,15 +121,30 @@ func (a *baseAction) saveResult(run flows.FlowRun, step flows.Step, name, value,
 }
 
 // helper to save a run result based on a webhook call and log it as an event
-func (a *baseAction) saveWebhookResult(run flows.FlowRun, step flows.Step, name string, webhook *flows.WebhookCall, status flows.CallStatus, logEvent flows.EventCallback) {
+func (a *baseAction) saveWebhookResult(run flows.FlowRun, step flows.Step, name string, webhook *flows.WebhookCall, status flows.CallStatus, bodyAsExtra bool, logEvent flows.EventCallback) {
 	input := fmt.Sprintf("%s %s", webhook.Method, webhook.URL)
 	value := strconv.Itoa(webhook.StatusCode)
 	category := webhookStatusCategories[status]
-	extra := utils.ExtractResponseJSON(webhook.Response)
+
+	var extra json.RawMessage
+	if bodyAsExtra && len(webhook.ResponseBody) < 10000 && json.Valid(webhook.ResponseBody) {
```

Comment:
so.. this is slightly different to what I said in #814.. but seems to me that not-setting extra if it's too big is no worse than erroring if it's too big.. both are runtime changes in behavior when the payload is big.. and at least this way.. 

- a user can use `@webhook` and not run into problems even if they've forgotten to uncheck "Save response as extra" on all the webhook actions
- that option on the action dialog can explain the 10K limit as well as advising that users use `@webhook`